internal/agent/upgrade.go

```diff
@@ -146,8 +146,8 @@ func generateUpgradeConfForCLIArgs(source, upgradeEntry string) (string, error)
 	// have access to that yet, we just set both uri values which shouldn't matter
 	// anyway, the right one will be used later in the process.
 	if source != "" {
-		upgradeConfig.Upgrade.RecoverySystem.URI = source
-		upgradeConfig.Upgrade.System.URI = source
+		upgradeConfig.Install.RecoverySystem.URI = source
```

Comment:
this feels wrong. This has worked with cli flags since forever, so it doesnt seem like it would be the source of the issue. The end config/spec has the proper sources so it seems wrong to me that this would be broken here?

Are we sure this doesnt break anything? IIRC our test use the cli flags to upgrade by setting a source, so this seems to be working as expected or our tests are fully broken.

I dont have anything else to proof this other than a feeling that if source cli flags work, this migth be working but I could be wrong...